src/main/java/com/example/Game.java

```diff
@@ -3,88 +3,30 @@
 import java.util.InputMismatchException;
 import java.util.Scanner;
 
-class Game {
+public class Game {
     private final Board board;
     private final Player player1;
     private final Player player2;
     private final Scanner scanner = new Scanner(System.in);
 
     public Game() {
-        System.out.println("Bored!Let's play a Game!!\uD83C\uDFB2");
-        System.out.println("Enter the grid size: (3 for 3x3, 4 for 4x4 etc)");
-        int size;
-        while (true)
-        {
-            try{
-                size = scanner.nextInt();
-                scanner.nextLine();
-                if(size>=3)
-                {
-                    break;
-                }
-                else {
-                    System.out.println("Grid size must be 3 or larger than 3");
-                }
-            }
-            catch (InputMismatchException e)
-            {
-                System.out.println("Invalid input! Please enter valid number");
-            }
-        }
-        board = new Board(size);
-
-        System.out.println("So, tell me what's your name?");
-        String playerName;
-        while (true)
-        {
-            playerName = scanner.nextLine();
-            if(!playerName.isEmpty())
-            {
-                break;
-            }
-            else {
-                System.out.println("Name cannot be empty. Please enter a valid name.");
-            }
-        }
+        System.out.println("Bored! Let's play a Game!! \uD83C\uDFB2");
 
+        int size = getGridSize();
+        board = new Board(size);
 
-        char playerSymbol;
-        while (true) {
-            System.out.println("Choose your symbol (X or O):");
-            playerSymbol = Character.toUpperCase(scanner.next().charAt(0));
-            if (playerSymbol == 'X' || playerSymbol == 'O') {
-                break;
-            } else {
-                System.out.println("Invalid symbol! Please choose X or O.");
-            }
-        }
-        player1 = new HumanPlayer(playerName, playerSymbol);
-
-        int difficultyLevel;
-        while (true) {
-            System.out.println("Choose difficulty level: 1 for EASY and 2 for HARD");
-            try {
-                difficultyLevel = scanner.nextInt();
-                scanner.nextLine();
-                if (difficultyLevel == 1 || difficultyLevel == 2) {
-                    break;
-                }
-                else
-                {
-                    System.out.println("Please enter 1 for EASY or 2 for HARD.");
-                }
-            } catch (InputMismatchException e) {
-                System.out.println("Invalid input. Please enter a valid number.");
-                scanner.next();
-            }
-        }
+        // Create player1 by using PlayerFactory and creating instance of HumanPlayerFactory
+        PlayerFactory humanFactory = new HumanPlayerFactory();
+        player1 = humanFactory.createPlayer();
```

Comment:
What is the purpose of factory,
If you are deciding object creation own your own?
You should ask user or there should be some sort of mechanism that decides object creation.

Reply:
That is already changed in newer commits. Please see newer commits I made.